net/tstun/wrap.go

```diff
@@ -615,63 +670,102 @@ func (c *natV4Config) selectSrcIP(oldSrc, dst netip.Addr) netip.Addr {
 	return oldSrc
 }
 
-// natV4ConfigFromWGConfig generates a natV4Config from nm.
-// If v4 NAT is not required, it returns nil.
-func natV4ConfigFromWGConfig(wcfg *wgcfg.Config) *natV4Config {
+// natConfigFromWGConfig generates a natConfig from nm.
+// If NAT is not required, it returns nil.
+func natConfigFromWGConfig(wcfg *wgcfg.Config) *natConfig {
 	if wcfg == nil {
 		return nil
 	}
-	nativeAddr := findV4(wcfg.Addresses)
-	if !nativeAddr.IsValid() {
+
+	nativeAddr4 := findV4(wcfg.Addresses)
+	nativeAddr6 := findV6(wcfg.Addresses)
+	if !nativeAddr4.IsValid() && !nativeAddr6.IsValid() {
 		return nil
 	}
+
 	var (
-		rt           table.RoutingTableBuilder
-		dstMasqAddrs map[key.NodePublic]netip.Addr
-		listenAddrs  set.Set[netip.Addr]
+		rt4, rt6                     table.RoutingTableBuilder
+		dstMasqAddrs4, dstMasqAddrs6 map[key.NodePublic]netip.Addr
+		listenAddrs4, listenAddrs6   set.Set[netip.Addr]
 	)
 
 	// When using an exit node that requires masquerading, we need to
```

Comment:
this comment seems to be out of place, it looks like it is about the var block above

Reply:
Nack, this is describing the need for the `exitNodeRequiresMasqX` variables, albeit confusingly.

Reply:
I think perhaps what I want is both, the outcome affects what is in `rt`.